src/PubSub/Subscription.php

```diff
@@ -0,0 +1,551 @@
+     * }
+     * ```
+     *
+     * @param  array $options
+     * @return bool
+     */
+    public function exists(array $options = [])
+    {
+        try {
+            $this->info($options);
+            return true;
+        } catch (GoogleNotFoundException $e) {
+            return false;
+        }
+    }
+
+    /**
+     * Get info on a subscription
+     *
+     * If the info is already cached on the object, it will return that result.
+     * To fetch a fresh result, use {@see Subscription::reload()}.
+     *
+     * ```php
+     * use Google\Cloud\ServiceBuilder;
+     *
+     * $cloud = new ServiceBuilder([
+     *      'projectId' => 'my-awesome-project'
+     * ]);
+     *
+     * $pubsub = $cloud->pubsub();
+     *
+     * $subscription = $pubsub->subscription('my-new-subscription');
+     * $info = $subscription->info();
+     * echo $info['name']; // `projects/my-awesome-project/subscriptions/my-new-subscription`
+     * ```
+     *
+     * @param  array $options Configuration options
+     * @return array Subscription data
+     */
+    public function info(array $options = [])
+    {
+        if (!$this->info) {
+            $this->reload($options);
+        }
+
+        return $this->info;
+    }
+
+    /**
+     * Retrieve info on a subscription from the API.
+     *
+     * To use the previously cached result (if it exists), use
+     * {@see Subscription::info()}.
+     *
+     * ```php
+     * use Google\Cloud\ServiceBuilder;
+     *
+     * $cloud = new ServiceBuilder([
+     *      'projectId' => 'my-awesome-project'
+     * ]);
+     *
+     * $pubsub = $cloud->pubsub();
+     *
+     * $subscription = $pubsub->subscription('my-new-subscription');
+     * $subscription->reload();
+     * $info = $subscription->info();
+     * echo $info['name']; // `projects/my-awesome-project/subscriptions/my-new-subscription`
+     * ```
+     *
+     * @param  array $options Configuration options
+     * @return array Subscription data
+     */
+    public function reload(array $options = [])
+    {
+        return $this->info = $this->connection->getSubscription($options + [
+            'subscription' => $this->name
+        ]);
+    }
+
+    /**
+     * Retrieve new messages from the topic.
+     *
+     * ```php
+     * use Google\Cloud\ServiceBuilder;
+     *
+     * $cloud = new ServiceBuilder([
+     *      'projectId' => 'my-awesome-project'
+     * ]);
+     *
+     * $pubsub = $cloud->pubsub();
+     *
+     * $subscription = $pubsub->subscription('my-new-subscription');
+     * $messages = $subscription->pull();
+     * foreach ($messages as $message) {
+     *      echo $message['data'];
+     * }
+     * ```
+     *
+     * @param  array     $options {
+     *      Configuration Options
+     *
+     *      @type bool $returnImmediately If set, the system will respond
+     *                 immediately. Otherwise, wait until new messages are
+     *                 available. Returns if timeout is reached.
```

Comment:
PubSub doesn't specify one. I was referring to a timeout set in the client. I removed that sentence because it was not clear.